### PR TITLE
fixed issue #94: create db folder for startup

### PR DESCRIPTION
**Pull request by Hamzsd**
Added code for creating folder and file if not already exists.

### CODE REVIEW

File: data/json.go [hunk 1]
```diff
@@ -152,6 +154,15 @@ type jsonConnector struct {
 
 func init() {
 	register("json", func(connStr string, l *common.Logger) (DataConnector, error) {
+
+		if _, err := os.Stat(filepath.Dir(connStr)); os.IsNotExist(err) {
```
Comment (CptPie) [inline data/json.go:158]: I think these changes are better located in the `newJsonConnector` function.

File: data/json.go [hunk 2]
```diff
@@ -152,6 +154,15 @@ type jsonConnector struct {
...
+			os.MkdirAll(filepath.Dir(connStr), 0777)
```
Comment (CptPie) [inline data/json.go:159]: I know that up until now we also used `0777`/`0775` for file permissions but do we really need the execute flag for these files? @zorchenhimer

**Review by CptPie — CHANGES_REQUESTED** [review]
First of all thank you for your contribution, even though i have a few points of to improve.
Aside from the comments below i've discovered a problem when compiling/executing your PR:
image
While the directory (`db`) and the file (`data.json`) get created just fine, that does not work out with our current json file reader. If the file exists, it expects a valid json format (even if its just an empty pair of `{}`), in your PR the file is just empty which causes the json reader to throw the displayed error.

File: data/json.go (hunk 2 above)
Comment (CptPie) [inline data/json.go:159]: os.MkdirAll returns an error - please dont throw it away, instead please handle it

File: data/json.go (hunk 1 above)
Comment (zorchenhimer) [reply data/json.go:158]: The JSON file is already created if it doesn't exist.  And as you've mentioned, an error is thrown when an empty JSON file is found.

File: data/json.go (hunk 2 above)
Comment (zorchenhimer) [reply data/json.go:159]: Directories need execute permissions or they won't be accessible.

**Comment by zorchenhimer** [comment]
As @CptPie mentioned, you should not be creating an empty JSON file.  It will be created on the first run.

Also, I'd prefer if this function was used unless there's a good reason not to: https://github.com/zorchenhimer/MoviePolls/blob/master/common/util.go#L16

**Comment by CptPie** [comment]
Since #96 did work on the same issue and got merged into master AND there was no response from the auther i'll close this now.

**Pull request closed without merging**